core/src/main/java/org/elasticsearch/common/xcontent/XContentParserUtils.java

```diff
@@ -35,6 +36,19 @@ private XContentParserUtils() {
     }
 
     /**
+     * Makes sure that current token is of type {@link XContentParser.Token#FIELD_NAME} and the the field name is equal to the provided one
+     * @throws ParsingException if the token is not of type {@link XContentParser.Token#FIELD_NAME} or is not equal to the given field name
+     */
+    public static void ensureFieldName(XContentParser parser, Token token, String fieldName) throws IOException {
+        ensureExpectedToken(Token.FIELD_NAME, token, parser::getTokenLocation);
+        String current = parser.currentName() != null ? parser.currentName() : "<null>";
```

Comment:
What do you mean? `current` is either the current tokens field name or "<null>" which I thought serves for printing the error message. I might be missing something.